packages/hub/src/session.ts

```diff
@@ -123,3 +134,120 @@ export class Session implements SessionInterface {
     });
   }
 }
+
+type releaseHealthAttributes = {
+  environment?: string;
+  release: string;
+};
+
+/**
+ * @inheritdoc
+ */
+export class SessionFlusher implements SessionFlusherInterface {
+  public readonly flushTimeout: number = 60;
+  private _pendingAggregates: { [key: number]: AggregationCounts } = {};
+  private _sessionAttrs: releaseHealthAttributes;
+  private _intervalId: ReturnType<typeof setInterval>;
+  private _isEnabled: boolean = true;
+  private _transport: Transport;
+
+  constructor(transport: Transport, attrs: releaseHealthAttributes) {
+    this._transport = transport;
+    // Call to setInterval, so that flush is called every 60 seconds
+    this._intervalId = setInterval(() => this.flush(), this.flushTimeout * 1000);
+    this._sessionAttrs = attrs;
+  }
+
+  /** Sends session aggregates to Transport */
+  public sendSessionAggregates(sessionAggregates: SessionAggregates): void {
+    if (!this._transport.sendSessionAggregates) {
+      logger.warn("Dropping session because custom transport doesn't implement sendSessionAggregates");
+      return;
+    }
+    this._transport.sendSessionAggregates(sessionAggregates).then(null, reason => {
+      logger.error(`Error while sending session: ${reason}`);
```

Comment:
Shouldn't this be a `.catch()` instead of a `.then()`? Otherwise we always log error when the promise is fulfilled.

Reply:
`catch` is just syntax sugar for `(null, reason => {})`, so it's the same :)

Reply:
ah TIL